### PR TITLE
test: Migrate src/app/devices/components to RTL

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Migrate all tests in src/app/devices/components to RTL

### CODE REVIEW

File: src/app/devices/components/DeviceHeaderForms/AddDeviceForm/types.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ export type AddDeviceInterface = {
 export type AddDeviceValues = {
   domain: Domain["name"];
   hostname: Device["hostname"];
-  interfaces: AddDeviceInterface[];
+  interfaces: (Omit<AddDeviceInterface, "subnet"> & { subnet: string })[];
```
Comment (petermakowski) [inline src/app/devices/components/DeviceHeaderForms/AddDeviceForm/types.ts:23]: I think we can modify the `AddDeviceInterface` directly and set the subnet to string, it's only used in this file.

File: src/app/devices/components/DeviceHeaderForms/DeviceHeaderForms.test.tsx [hunk 2]
```diff
@@ -1,31 +1,50 @@
...
+      <DeviceHeaderForms
+        devices={[]}
+        headerContent={{ view: DeviceHeaderViews.ADD_DEVICE }}
+        setHeaderContent={jest.fn()}
+      />,
+      { store }
     );
 
-    expect(wrapper.find("AddDeviceForm").exists()).toBe(true);
+    expect(screen.getByLabelText("Add device")).toBeInTheDocument();
```
Comment (petermakowski) [inline src/app/devices/components/DeviceHeaderForms/DeviceHeaderForms.test.tsx:48]: This returns the "Add device" button, to get the form you'd need to use `screen.getByRole("form", { name: "Add device" })`

**Review by petermakowski — APPROVED** [review]
LGTM, would be good to make some small changes.

**Comment by ndv99** [comment]
There ya go, made those changes

**Pull request merged**